include/robots/rc_car_bot.h

```diff
@@ -4,30 +4,58 @@
 // BoB robotics includes
 #include "common/i2c_interface.h"
 #include "robots/ackermann.h"
+#include "robots/rc_car_common.h"
 #include "third_party/units.h"
 
 // Standard C includes
 #include <cmath>
 #include <cstdint>
 
 // Standard C++ includes
+#include <utility>
 #include <vector>
 
 namespace BoBRobotics {
 namespace Robots {
 using namespace units::literals;
 
+//----------------------------------------------------------------------------
+// BoBRobotics::Robots::PassiveRCCarBot
+//----------------------------------------------------------------------------
+//! An interface for RC car robots which are being passively driven by the
+//! remote control. Don't cast to RCCarBot otherwise you'll get slicing issues!
+class PassiveRCCarBot
+{
+    using degree_t = units::angle::degree_t;
+
+public:
+    PassiveRCCarBot(const char *path = I2C_DEVICE_DEFAULT);
+
+    //! Read speed and turn values from remote control
+    std::pair<float, degree_t> readRemoteControl();
+
+    constexpr static degree_t TurnMax{ 35 };
+
+private:
+    BoBRobotics::I2CInterface m_I2C; // i2c interface
+
+protected:
+    PassiveRCCarBot(const char *path, RCCar::State initialState);
+    void setState(RCCar::State state);
+    void writeMessage(const RCCar::Message &msg);
+}; // PassiveRCCarBot
+
 //----------------------------------------------------------------------------
 // BoBRobotics::Robots::RCCarBot
 //----------------------------------------------------------------------------
 //! An interface for 4 wheeled, Arduino-based robots developed at the University of Sussex
 class RCCarBot final
-  : public Ackermann
+  : public Ackermann, public PassiveRCCarBot
```

Comment:
Why not solve this multiple inheritance and the forementioned slicing issue by having ``RCCarBot`` own a ``PassiveRCCarBot`` and prodding it rather than inheriting from it (you'll also need to a little befriending)